src/configureStore.ts

```diff
@@ -72,7 +76,7 @@ export interface ConfigureStoreOptions<S = any, A extends Action = AnyAction> {
    * function (either directly or indirectly by passing an object as `reducer`),
    * this must be an object with the same shape as the reducer map keys.
    */
-  preloadedState?: DeepPartial<S>
+  preloadedState?: DeepPartial<PS>
```

Comment:
What you said about users specifying an explicit type for `PS` that has properties not in `S` got me thinking about how to get rid of the `PS` type, and I figured out that I could lower the "inference priority", so to speak, of the preloaded state by adding a bit of indirection with a ternary that will always return `S`.

```js
export function configureStore<S = any, A extends Action = AnyAction>(  // no more `PS` type
  options: ConfigureStoreOptions<S, A>
): Store<S, A> {
//...
}

//...

export interface ConfigureStoreOptions<S = any, A extends Action = AnyAction> {  // no more `PS` type
// ...
preloadedState?: DeepPartial<S extends any ? S : S>;  // <- this ternary
//...
}
```

The `S extends any ? S : S` ternary will always result in `S` but it prevents `preloadedState` from overiding the state inferred from the reducer object.

It looks janky, but no more `PS` type allowing users to potentially shoot themselves in the foot.

Reply:
Also it means you no longer have to type cast `preloadedState as DeepPartial<S>,` in `createStore`

Reply:
@Dudeonyx Nice hack! I applied that version (including an explanatory comment).